tests/tools/test_processes.py

```diff
@@ -2,56 +2,547 @@
+    return content_list[0].text
+
+
 class TestProcesses:
     """Test process management tools."""
 
     async def test_list_processes_returns_string(self):
         """Test that list_processes returns a string."""
-        result = await processes.list_processes()
-        assert isinstance(result, str)
-        assert len(result) > 0
+        result = await mcp.call_tool("list_processes", {})
+        text = _get_text_from_mcp_result(result)
+        assert isinstance(text, str)
+        assert len(text) > 0
 
     async def test_list_processes_contains_process_info(self):
         """Test that list_processes contains process information."""
-        result = await processes.list_processes()
+        result = await mcp.call_tool("list_processes", {})
+        text = _get_text_from_mcp_result(result)
 
         # Should contain process-related keywords
-        assert "pid" in result.lower() or "process" in result.lower()
+        assert "pid" in text.lower() or "process" in text.lower()
         # Should contain resource usage info
-        assert "cpu" in result.lower() or "memory" in result.lower() or "mem" in result.lower()
+        assert "cpu" in text.lower() or "memory" in text.lower() or "mem" in text.lower()
+
+    async def test_list_processes_shows_top_100(self):
+        """Test that list_processes limits output to top 100 processes."""
+        result = await mcp.call_tool("list_processes", {})
+        text = _get_text_from_mcp_result(result)
+        assert "Top 100" in text
+        assert "Total processes:" in text
+
+    async def test_list_processes_handles_no_access_denied(self, mocker):
+        """Test that list_processes handles AccessDenied exceptions."""
+        mock_proc = mocker.Mock()
+        mock_proc.info = {"pid": 1, "name": "test"}
+
+        def process_iter_side_effect(*args, **kwargs):
+            # Return one good process and one that raises AccessDenied
+            yield mock_proc
+            mock_bad = mocker.Mock()
+            mock_bad.info
+            type(mock_bad).info = mocker.PropertyMock(side_effect=psutil.AccessDenied())
+            yield mock_bad
+
+        mocker.patch("psutil.process_iter", side_effect=process_iter_side_effect)
+        mocker.patch("psutil.process_iter", return_value=[mock_proc])
+
+        result = await mcp.call_tool("list_processes", {})
+        text = _get_text_from_mcp_result(result)
+        assert isinstance(text, str)
+        assert "Running Processes" in text
+
+    async def test_list_processes_exception_handling(self, mocker):
+        """Test that list_processes handles general exceptions."""
+        mocker.patch("psutil.process_iter", side_effect=Exception("Test error"))
+
+        result = await mcp.call_tool("list_processes", {})
+        text = _get_text_from_mcp_result(result)
+        assert "Error listing processes" in text
+        assert "Test error" in text
 
     async def test_get_process_info_with_current_process(self):
```

Comment:
It seems like many of these tests can be combined into a single test with different parameters.